operators/src/pro/cache/error.rs

```diff
@@ -0,0 +1,22 @@
+use snafu::Snafu;
+
+use crate::error::Error;
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility(pub(crate)))]
+#[snafu(context(suffix(false)))]
+pub enum CacheError {
+    CacheMustBeLargerOrEqualThanLandingZone,
+    NotEnoughSpaceInLandingZone,
+    NotEnoughSpaceInCache,
+    QueryNotFoundInLandingZone,
+    InvalidRasterDataTypeForInsertion,
+}
+
+impl From<CacheError> for Error {
+    fn from(e: CacheError) -> Self {
+        Error::QueryingProcessorFailed {
+            source: Box::new(e),
+        }
+    }
```

Comment:
Are these errors where the engine fails?

Reply:
you are right, these are internal errors that should not interfere with what the user gets to see. However, the functions should not panic, and have to return something if something goes wrong. So shall I change the return type of the TileCache functions to Options instead of Results then?

Reply:
I guess it is good for operations and debugging, that there are distinct errors.
Could we then just not convert it into Errors that we hand out? This way, we ensure that they stay internally.

Reply:
2bfd416ea160135be0a8c7d68b6d550e40eaa7a3